core/src/main/java/org/infinispan/util/concurrent/BlockingManager.java

```diff
@@ -279,4 +315,7 @@ interface BlockingExecutor {
        */
       <V> CompletionStage<V> supply(Supplier<V> supplier, Object traceId);
    }
+
+   interface ScheduledBlockingCompletableStage<V> extends ScheduledCompletableStage<V>, ScheduledFuture<V> {
```

Comment:
I am personally not a fan of exposing this as a Future here as I am trying to make all these methods return a Stage. But there is no interface I know of that exposes a cancellable non blocking computation so I guess it is fine.